### ISSUE TITLE
Leading spaces and auto-indent incorrectly converts into tabs

### ISSUE THREAD

**Issue by mmoult** [notepad-plus-plus/notepad-plus-plus#13130]
### Description of the Issue

Issue [#2215](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/2215) was closed even though the problem persists. I am opening this new issue in its stead. In the issue, there are two closely-related problems:
1. leading spaces in a line are converted into tabs upon enter
2. auto-indent uses tabs instead of duplicating the leading spacing of the previous line

I know that there have been other issues to enable spaces by default in Python files, but I feel like they are bandages over the underlying problem. If more languages like Python became popular, we would have to create similar issues for each one. Not fixing this issue invites future problems.

As @baarney said so succinctly:
"It seems so simple in principle:
- If I press -tab- I want a tab.
- If I press -space- I want a space.
- If I press -enter- at the end of a line, I want the next line indented with exactly the same leading whitespace chars as the previous line."

In some of my .txt files, I use tabs on some lines and spaces on others as I see fit. Therefore, it is not sufficient to merely convert all tabs to spaces. I echo @aaronmcdavid, "I expected NotePad++ to leave my spaces and tabs alone. I know what's best for my file." Clearly I am not the only one with this sentiment.

### Steps to Reproduce the Issue

0. Open an instance of Notepad++
1. On a new line in the main window, type the following:
`«5 spaces»\tThis line has leading spaces and a manual tab«Enter»`
2. Get a completely new line by hitting enter again and using backspace (twice)
4. Type the following:
«5 spaces»Auto-indent should duplicate the leading spaces below:«Enter»

### Expected Behavior

1. I expected that Notepad++ would not change the leading spacing of a line after pressing enter.
2. I expected that Notepad++ would use the duplicate the leading whitespace from the previous line as auto-indent for the next

### Actual Behavior

After step 1, you can observe that the five spaces and a tab on the first line you typed were converted into two tabs. If this is a "feature", then there should be a way to turn it off. If this "feature" is to be preserved, I don't think it should be on by default. It is unintuitive for a text editor to retroactively edit characters in a line upon hitting enter.

After step 3, you will notice that the leading whitespace on the current line is a tab and a space (instead of 5 spaces). In this case, the leading whitespace of the previous line was not edited (thankfully) and as such, the leading whitespace is different between the last line and the current line.

### Debug Information

Notepad++ v8.4.9   (64-bit)
Build time : Jan 27 2023 - 03:11:16
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Program Files\Notepad++\change.log" 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 (64-bit) 
OS Version : 22H2
OS Build : 22621.1194
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

**Comment by westyles** [comment]
Hello.
Have you found a solution to this problem of converting all spaces to tabs?
I don't like the current indentation behavior of npp8.7 either.

What you describe should have been the default at all times!
Because if you enable the conversion of indents to Spaces, it's not possible to type Tab anywhere at all!

And if you make the indentation setting as Tabs, then all initial spaces equal to the size of the indentation setting are converted to Tabs, and it's horrible.
And there's no option for the usual intuitive behavior, other than to disable auto indentation altogether.
Need the ability to simply duplicate the previous initial indentation, i.e. auto indent with no conversions at all!
